app/controllers/guide_article_drafts_controller.ts

```diff
@@ -0,0 +1,286 @@
+      return;
+    }
+
+    // For actions without IDs (index, store), check class-level permission
+    const slugMap: Record<Action, string> = {
+      index: "read",
+      show: "read",
+      store: "create",
+      update: "update",
+      destroy: "destroy",
+      relationIndex: "read",
+      oneToManyRelationStore: "update",
+      manyToManyRelationAttach: "update",
+      manyToManyRelationDetach: "update",
+    };
+    const slug = slugMap[action];
+
+    const allowed = await user?.hasPermission?.(slug, this.model);
+    if (allowed !== true) {
+      throw new ForbiddenException();
+    }
+  }
+
+  /**
+   * Override to disable the base permission check since we handle it in authenticate()
+   */
+  protected requiredPermissionFor(_action: Action) {
+    return null;
+  }
+
+  protected async authorizeById(
+    http: HttpContext,
+    action: Action,
+    ids: { localId?: number | string },
+  ) {
+    const user = http.auth.user as unknown as
+      | {
+          hasRole?: (slug: string) => Promise<boolean>;
+          hasPermission?: (
+            action: string,
+            target?: unknown,
+          ) => Promise<boolean>;
+        }
+      | undefined;
+    const isAdmin =
+      (await user?.hasRole?.("solvro_admin")) === true ||
+      (await user?.hasRole?.("admin")) === true;
+    if (isAdmin) {
+      return;
+    }
+
+    const { localId } = ids;
+    const maybeId = localId;
+    const draftId = typeof maybeId === "string" ? Number(maybeId) : maybeId;
+    if (draftId === undefined) {
+      return;
+    }
+
+    const slugMap: Record<string, string> = {
+      index: "read",
+      show: "read",
+      store: "create",
+      update: "update",
+      destroy: "destroy",
+      relationIndex: "read",
+      oneToManyRelationStore: "update",
+      manyToManyRelationAttach: "update",
+      manyToManyRelationDetach: "update",
+    };
+    const slug = slugMap[action] ?? "read";
+
+    const allowed = await user?.hasPermission?.(slug, this.model);
+    if (allowed === true) {
+      return;
+    }
+
+    const instance = await this.model.find(draftId);
+    if (instance === null) {
+      throw new ForbiddenException();
+    }
+    const has = await user?.hasPermission?.(slug, instance);
+    if (has !== true) {
+      throw new ForbiddenException();
+    }
+  }
+
+  /**
+   * When linking to an existing article, require per-model permission on the original article.
+   *
+   * Note: Multiple drafts can be created for the same article. This is intentional to allow:
+   * - Different users to propose different changes to the same article
+   * - A user to create multiple draft versions before deciding which to submit
+   * - Parallel editing workflows
+   *
+   * The solvro_admin can approve/reject drafts individually, choosing which changes to apply.
+   */
+  protected async storeHook(ctx: {
+    http: HttpContext;
+    request: PartialModel<typeof GuideArticleDraft>;
+  }) {
```

Comment:
unaddressed concern from previous review: what if multiple drafts are created for the same article?